ocaml/lambda/translcore.ml

```diff
@@ -1792,7 +1793,9 @@ and transl_record ~scopes loc env mode fields repres opt_init_expr =
   let size = Array.length fields in
   (* Determine if there are "enough" fields (only relevant if this is a
      functional-style record update *)
-  let no_init = match opt_init_expr with None -> true | _ -> false in
+  let (no_init, ubr) = match opt_init_expr with
+    | None -> (true, MayBePushedDown)
```

Comment:
This is a bit scary. But it's OK: the `ubr` is used only in the `Kept` case which can happen only when this is `Some`. Still, I think I'd like the extraction of the `ubr` to happen closer to where it's used, so that there doesn't have to be a `MayBePushedDown` here. That means that the `None` case might have an `Misc.fatal_error` or something -- that feels safer than an under-motivated `MayBePushedDown`.

Reply:
I agree. I could turn the `if no_init || ..` statement into a case-split on `opt_init_expr`. This would eliminate the boolean-blindness of `no_init` and make the unique_barrier only available in the first branch where it is actually used.